### PR TITLE
Nick/utility api

### PR DESCRIPTION
**Pull request by niwong**
[//]: # "These comments meant for your reference, they are invisible and don't need to be deleted"

## What's new in this PR?

• Initialized `utility-api.js` file that performs an API call to Utility API retrieving a given electricity meter's latest bill and returns an object of the form:
```
{ netUsage: -162.2869, ebce_rebate: 11.16 }
```

• Edited `.gitignore` following best practices for node applications' standard `.gitignore`.
• Added `UTILITY_API_KEY` to the `.env.example`.

[//]: # "Describe what's new in this PR"

### How to Review

• Add Utility API Key to your local `.env` file.
• Run `node index.js`.
• Review `utils/utility-api.js` to verify I'm taking the right numbers and performing the correct addition to calculate the `netUsage` and `ebce_rebate`.

[//]: # 'The order in which to review files and what to expect when testing locally'

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc"

### Next steps

[//]: # "What doesn't work yet, what's NOT in this PR/has to be done "

### Tests Performed, Edge Cases

[//]: # 'If you made changes/additions to `airtable.js` make sure you update/add'
[//]: # '[at least 1] corresponding unit tests in `airtable.spec.js`'

### Screenshots

[//]: # "Add screenshots!!! If you'd like"

CC: @dfangshuo

### CODE REVIEW

File: utils/utility-api.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+	const line_items = responseToJson.bills[0]["line_items"];
+	const rebate_item = line_items.filter(function(element) {
+		return element.name === 'Credited to (Debited from) NEM Balance'
+	})
+
+	const rebate = rebate_item[0].cost
+
+	var returnObj = {
+		netUsage: netUsage,
```
[the remark this answers is not in the document]
Comment (aivantg) [reply utils/utility-api.js]: On this same vein, I think you can name the rebate variable `ebceRebate` and then use this instead

[the remark this answers is not in the document]
Comment (aivantg) [reply utils/utility-api.js]: Can you also return the start and end date of the bill?

File: utils/utility-api.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+const getLatestBill = async (meterUID) => {
+
+	var meter_uid = '?meters=' + meterUID;
+	var latest = '&limit=1&order=latest_first';
+
+	var url = bills_base_url + meter_uid + latest;
+	const response = await fetch(url, options);
+	const responseToJson = await response.json();
+	const pge_details = responseToJson.bills[0]["pge_details"];
+	const netUsage = pge_details.consumption + pge_details.net_generation;
```
[the remark this answers is not in the document]
Comment (niwong) [reply utils/utility-api.js]: Yup! `"bill_total_volume"` looks like the right key from looking again at a sample bill object. I confirmed that it's equal to `consumption + net_generation`.

File: utils/utility-api.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+	const line_items = responseToJson.bills[0]["line_items"];
+	const rebate_item = line_items.filter(function(element) {
```
[the remark this answers is not in the document]
Comment (niwong) [reply utils/utility-api.js]: Oooh some ES6 ninja-work. Thank you for sharing. Implemented!
